pkg/s3select/select.go

```diff
@@ -363,81 +378,57 @@ func processColNameLiteral(record string, reqColNames []string, myFunc SelectFun
 	return rowStr, nil
 }
 
-// aggregationFunctions is a function which performs the actual aggregation
+// aggregationFns is a function which performs the actual aggregation
 // methods on the given row, it uses an array defined in the main parsing
 // function to keep track of values.
-func aggregationFunctions(counter int, filtrCount int, myAggVals []float64, storeReqCols []string, storeFunctions []string, record string) error {
-	for i := 0; i < len(storeFunctions); i++ {
-		if storeFunctions[i] == "" {
-			i++
-		} else if storeFunctions[i] == "count" {
-			myAggVals[i]++
-		} else {
-			// If column names are provided as an index it'll use this if statement instead of the else/
+func aggregationFns(counter int, filtrCount int, aggVals []float64, storeReqCols []string, storeFns []string, record []byte) error {
+	for i, storeFn := range storeFns {
+		switch storeFn {
+		case "":
+			continue
+		case "count":
+			aggVals[i]++
+		default:
+			// Column names are provided as an index it'll use
+			// this if statement instead.
 			var convAggFloat float64
 			if format.IsInt(storeReqCols[i]) {
-				myIndex, _ := strconv.Atoi(storeReqCols[i])
-				convAggFloat, _ = strconv.ParseFloat(jsonValue(string("_"+strconv.Itoa(myIndex)), record), 64)
-
+				index, _ := strconv.Atoi(storeReqCols[i])
```

Comment:
It looks like the two following lines can be converted to
`convAggFloat = gjson.GetBytes(record, "_"+storeReqCols[i]).Float()`

Reply:
This is not safe to do - that is why have to do it the way it is right now.